goss_config.go

```diff
@@ -102,65 +166,7 @@ func interfaceMap(slice interface{}) map[string]interface{} {
 func mergeGoss(g1, g2 GossConfig) GossConfig {
 	g1.Gossfiles = nil
 
-	for k, v := range g2.Files {
-		g1.Files[k] = v
-	}
-
-	for k, v := range g2.Packages {
-		g1.Packages[k] = v
-	}
-
-	for k, v := range g2.Addrs {
-		g1.Addrs[k] = v
-	}
-
-	for k, v := range g2.Ports {
-		g1.Ports[k] = v
-	}
-
-	for k, v := range g2.Services {
-		g1.Services[k] = v
-	}
-
-	for k, v := range g2.Users {
-		g1.Users[k] = v
-	}
-
-	for k, v := range g2.Groups {
-		g1.Groups[k] = v
-	}
-
-	for k, v := range g2.Commands {
-		g1.Commands[k] = v
-	}
-
-	for k, v := range g2.DNS {
-		g1.DNS[k] = v
-	}
-
-	for k, v := range g2.Processes {
-		g1.Processes[k] = v
-	}
-
-	for k, v := range g2.KernelParams {
-		g1.KernelParams[k] = v
-	}
-
-	for k, v := range g2.Mounts {
-		g1.Mounts[k] = v
-	}
-
-	for k, v := range g2.Interfaces {
-		g1.Interfaces[k] = v
-	}
-
-	for k, v := range g2.HTTPs {
-		g1.HTTPs[k] = v
-	}
-
-	for k, v := range g2.Matchings {
-		g1.Matchings[k] = v
-	}
+	g1.Merge(g2)
```

Comment:
not clear to me why here `g1.Gossfiles` is set to nil, does not neccesary seems desirable in a generic merge, so I left it here to preserve existing behaviour and didnt put it on the new `Merge()` function.

Reply:
@ripienaar this is interesting. I tranced this code to the [conception](https://github.com/aelsabbahy/goss/pull/2) and it's still a mystery why it's needed. 😞

Reply:
Not sure to be honest, maybe the pointer was causing some panic or something back then.

I'm not sure there's any need for it now. *shrug*